### PR TITLE
Proper phase/finalizer logic in federated namespace controller

### PR DESCRIPTION
**Pull request by mwielgus**
Part 1 of delete logic cleanup.
Part 2 will contain namespace content deletion and tests.

cc: @quinton-hoole @kubernetes/sig-cluster-federation

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/31438)

### CODE REVIEW

File: federation/pkg/federation-controller/namespace/namespace_controller.go [hunk 1]
```diff
@@ -302,3 +298,51 @@ func (nc *NamespaceController) reconcileNamespace(namespace string) {
...
+func (nc *NamespaceController) delete(namespace *api_v1.Namespace) {
+	// Set Terminating status.
+	updatedNamespace := &api_v1.Namespace{
+		ObjectMeta: namespace.ObjectMeta,
+		Spec:       namespace.Spec,
+		Status: api_v1.NamespaceStatus{
+			Phase: api_v1.NamespaceTerminating,
+		},
+	}
+	if namespace.Status.Phase != updatedNamespace.Status.Phase {
```
Comment (wojtek-t) [inline federation/pkg/federation-controller/namespace/namespace_controller.go]:
```
if namespace.Status.Phase != api_v1.NamespaceTerminating {
```

**Comment by wojtek-t** [comment]
Minor nit - generally lgtm

File: federation/pkg/federation-controller/namespace/namespace_controller.go (hunk 1 above)
Comment (mwielgus) [inline federation/pkg/federation-controller/namespace/namespace_controller.go]: done.

**Comment by mwielgus** [comment]
all comments fixed, applying lgtm

**Pull request merged**